test/negotiation/utils/NegotiationTest.js

```diff
@@ -0,0 +1,473 @@
+  });
+  bar.update(100);
+  bar.stop();
+  console.log('');
+  browserInstalled = true;
+};
+
+let browser;
+
+before(async function() {
+  this.timeout(30000);
+  if (!browserInstalled) {
+    this.timeout(300000);
+    await installBrowser();
+  }
+  browser = await puppeteer.launch({
+    headless: true,
+    executablePath: revisionInfo.executablePath,
+    args: [
+      '--use-fake-ui-for-media-stream',
+      '--use-fake-device-for-media-stream',
+    ]
+  });
+  const internalPage = await browser.newPage();
+  await internalPage.goto(`chrome://webrtc-internals`);
+});
+
+after(async function() {
+   await browser.close();
+});
+
+
+const describeNegotiationTest = function(title, test) {
+  describe(title, function() {
+    this.timeout(30000);
+
+    const ctx = {
+      clientStreams: {},
+      erizoStreams: {},
+      expectedClientSdpVersion: 2,
+      expectedErizoSdpVersion: 0,
+    };
+
+    let page;
+    const erizoConns = {};
+
+    ctx.getFirstClientStream = function() {
+      const streamIds = Object.keys(ctx.clientStreams);
+      if (streamIds.length > 0) {
+        return ctx.clientStreams[streamIds[0]];
+      }
+      return;
+    };
+
+    ctx.getFirstRemoteStream = function() {
+      const streamIds = Object.keys(ctx.erizoStreams);
+      if (streamIds.length > 0) {
+        return ctx.erizoStreams[streamIds[0]];
+      }
+      return;
+    };
+
+    ctx.createClientStream = async function() {
+      const stream = new ClientStream(page);
+      ctx.clientStreams[stream.id] = stream;
+      await stream.init();
+      await stream.waitForAccepted();
+      await stream.getLabel();
+      return stream;
+    };
+
+    ctx.deleteClientStream = async function(stream) {
+      await ctx.clientStreams[stream.id].remove();
+      delete ctx.clientStreams[stream.id];
+    };
+
+    ctx.createRemoteStream = async function() {
+      const id = parseInt(Math.random() * 1000, 0);
+      const stream = { id, label: id.toString(), audio: true, video: true, addedToConnection: false };
+      ctx.erizoStreams[id] = stream;
+      return stream;
+    };
+
+    ctx.deleteRemoteStream = async function(erizoStream) {
+      delete ctx.erizoStreams[stream.id];
+    };
+
+    ctx.createClientConnection = async function() {
+      const connectionId = parseInt(Math.random() * 1000, 0);
+      const conn = new ClientConnection(page, connectionId);
+      ctx.client = conn;
+      await conn.init();
+      return conn;
+    };
+
+    ctx.createErizoConnection = async function() {
+      const connectionId = parseInt(Math.random() * 1000, 0);
+      const conn = new ErizoConnection(connectionId);
+      await conn.init();
+      ctx.erizo = conn;
+      return conn;
+    };
+
+    after(async function() {
+      await page.close();
+      ctx.erizo.removeAllStreams();
+      ctx.erizo.close();
+    });
+
+    before(async function() {
+      const htmlPath = '/Users/jcague/development/licode/extras/basic_example/public/index.html';
```

Comment:
Consider making this dependent of the path of the process, or a similar thing to avoid having to update the path here.

Reply:
good catch!